### PR TITLE
Batching on custom interval

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Don't send 50 logs immediately if transmission interval is custom

## Related PRs or issues

[AB#61886](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/61886)

### CODE REVIEW

**Comment by Jamminroot** [comment]
Only approved current changes since you send link to review - I do see PR is not finished yet

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java [hunk 1]
```diff
@@ -770,27 +769,52 @@ synchronized void checkPendingLogs(@NonNull GroupState groupState) {
...
-            long delay = groupState.mBatchTimeInterval;
-            if (groupState.mBatchTimeInterval > MINIMUM_TRANSMISSION_INTERVAL) {
-                long now = System.currentTimeMillis();
-                long startTimer = SharedPreferencesManager.getLong(START_TIMER_PREFIX + groupState.mName);
-                if (startTimer == 0) {
-                    SharedPreferencesManager.putLong(START_TIMER_PREFIX + groupState.mName, now);
-                } else if (startTimer > now) {
-                    SharedPreferencesManager.remove(START_TIMER_PREFIX + groupState.mName);
+        long batchTimeInterval = groupState.mBatchTimeInterval;
+        AppCenterLog.debug(LOG_TAG, "checkPendingLogs(" + groupState.mName + ") pendingLogCount=" + pendingLogCount + " batchTimeInterval=" + batchTimeInterval);
```
Comment (annakocheshkova) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java]: String.format

File: sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java [hunk 2]
```diff
@@ -770,27 +769,52 @@ synchronized void checkPendingLogs(@NonNull GroupState groupState) {
...
+                SharedPreferencesManager.remove(START_TIMER_PREFIX + groupState.mName);
+                AppCenterLog.debug(LOG_TAG, "Invalid timer value for " + groupState.mName + " channel has been removed.");
+            }
+
+            /* If the interval is over. */
+            else if (startTimer + batchTimeInterval < now) {
+
+                /* Send all logs without any additional timers. */
+                if (pendingLogCount > 0) {
+                    triggerIngestion(groupState);
```
Comment (annakocheshkova) [inline sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java]: I would add some logging here, too

Comment (MatkovIvan) [reply sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java]: There is a log inside

**Pull request merged**